### PR TITLE
swi-prolog: Updated to version 9.0.0

### PR DESCRIPTION
**Pull request by JanWielemaker**
#### Description

Start new stable/development cycle.   Updated dependencies.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.4 21F79 arm64
Xcode 13.4.1 13F100

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: lang/swi-prolog/Portfile [hunk 1]
```diff
@@ -2,11 +2,12 @@
 
 PortSystem          1.0
 PortGroup           cmake 1.1
+PortGroup           compiler_blacklist_versions 1.0
```
Comment (neverpanic) [inline lang/swi-prolog/Portfile:5]: Why did you add the `compiler_blacklist_versions` port group when you're not blacklisting any compiler?

Comment (JanWielemaker) [reply lang/swi-prolog/Portfile:5]: Good question.   I simply synchronized with the development version.   Git blame tells me

     623d210033af (Renee Otten    2022-03-08 11:25:01 -0500  5) PortGroup           compiler_blacklist_versions 1.0

with as commit message _"blacklist clang < 900"_ and in the portfile comment

    to avoid the following problem: fatal error: 'threads.h' file not found
+compiler.blacklist-append {clang < 900}

Comment (pmetzger) [reply lang/swi-prolog/Portfile:5]: Push a new version perhaps?

Comment (JanWielemaker) [reply lang/swi-prolog/Portfile:5]: Considering @reneeotten made this change to the devel version rather recently I'd first like to see his response.    I'm not much of a Mac user ...

Comment (reneeotten) [reply lang/swi-prolog/Portfile:5]: well that was needed because of the error message you show above, if that is still the case for older macOS versions you should besides adding the PortGroup of course also retain the block:
```
# to avoid the following problem: fatal error: 'threads.h' file not found
compiler.blacklist-append {clang < 900}
```

This issue will not show up in the CI as for the latest macOS versions the clang version is new enough. I suspect the issue is still present so you likely will still need this. If you don't have an older macOS version to test this on (I don't have that...) then you could consider committing it without and adding it back if there failure at earlier OSes still persists. 

@JanWielemaker can you please please make sure to expand the tabs, the CI warns you about this.

Comment (JanWielemaker) [reply lang/swi-prolog/Portfile:5]: The lack of `threads.h` was also a problem when building using gcc-12 on the M1 and as a result there is now a CMake check for this file and some `#ifdef` logic to work around it.    So, I think we can  remove the `compiler_blacklist_versions`.   

Should I make a PR for this, or simply do this with the next release?

B.t.w.  gcc-12 gives about 30% better performance on Intel and about 8% on the M1 compared to the latest Clang.    What is the Macports best practice to make a gcc-based version available?

Comment (reneeotten) [reply lang/swi-prolog/Portfile:5]: thanks for the response @JanWielemaker - just remove the PortGroup in both ports as part of this PR please. There certainly is a way to have GCC compilers added as variant to build to port; I cannot tell you from the top of my head to do that. But there are many other ports who do it, so you can look at examples or I can take a stab at it at some later time.
